libslide/src/partial_evaluator/types.rs

```diff
@@ -2,12 +2,14 @@ use crate::evaluator_rules::RuleName;
 
 pub struct EvaluatorContext {
     pub(crate) rule_blacklist: Vec<RuleName>,
+    pub(crate) always_flatten: bool,
 }
 
 impl Default for EvaluatorContext {
     fn default() -> Self {
         Self {
             rule_blacklist: vec![],
+            always_flatten: true,
         }
     }
 }
```

Comment:
I think this default constructor should take a parameter to know to flatten rather then set to true but this can be down in another PR.

Reply:
The default ctor takes no parameters because it is the "default". There is a builder method to set this flag in the EvaluatorContext impl.